### PR TITLE
Speed up load time

### LINKED ISSUE #131: Refactor AdminClubPage getData()
Refactor the function getData(), so that it only calls setState once. 
Right now is calling setState multiple time asynchronously, which could led to issues.

### PR DESCRIPTION
**Pull request by steven-solar**
This PR alters the way club related pages fetch data to utilize the asynchronous nature of fetches to our advantage, parallelizing independent fetches and erasing possible issues with overriding state.

### CODE REVIEW

**Review by MauroGuerrero20 — APPROVED** [review]
LGTM

File: capstone-frontend/src/components/MyClubs.jsx [hunk 1]
```diff
@@ -16,47 +16,33 @@ class MyClubs extends Component {
...
-          .catch(e => console.log(e));
-      }
-
-      c.ownerName = owner.fullName;
-      c.bookTitle = book.title;
-    }
-    this.setState({ clubs: allClubs, fetchingClubs: false });
+          .then(book => book.authors && book.authors.length
+            ? c.bookTitle = book.title
+            : c.bookTitle = 'Nothing yet'
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/MyClubs.jsx:40]: I like the handling here when there isn't a book!

File: capstone-frontend/src/components/MyClubs.jsx [hunk 2]
```diff
@@ -16,47 +16,33 @@ class MyClubs extends Component {
...
-    }
-
-    let allClubs = memberClubs.concat(ownerClubs.filter((item) => memberClubs.indexOf(item) < 0));
-    let c;
-    for await (c of allClubs) {
-      let owner = await fetch(`/api/user?id=${c.ownerID}`)
+    let memberClubs = [];
+    let ownerClubs = [];
+    await Promise.all([
+      fetch(`/api/clubs?memberIDs=${window.localStorage.getItem("userID")}`)
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/MyClubs.jsx]: This looks much cleaner!

**Review by antoniolinhart — APPROVED** [review]
This should make the club page much faster!

**Review by VivekShankar — APPROVED** [review]
LGTM - nice fix!

File: capstone-frontend/src/components/MyClubs.jsx (hunk 1 above)
Comment (steven-solar) [reply capstone-frontend/src/components/MyClubs.jsx:40]: Thanks!

File: capstone-frontend/src/components/MyClubs.jsx (hunk 2 above)
Comment (steven-solar) [reply capstone-frontend/src/components/MyClubs.jsx]: Thank you!

**Pull request merged**